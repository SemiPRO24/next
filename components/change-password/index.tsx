import React, { Component } from 'react';

import { auth } from '../../firebase/index';

type ChangePasswordState = {
  password: string;
  confirmPassword: string;
  error: any;
};

const INITIAL_STATE: ChangePasswordState = {
  password: '',
  confirmPassword: '',
  error: null,
};

export class ChangePassword extends Component<{}, ChangePasswordState> {
  state = {
    ...INITIAL_STATE,
  };

  onSubmit = (e: any): void => {
    const { password } = this.state;

    auth
      .doPasswordUpdate(password)
      .then(() => {
        this.setState(() => ({ ...INITIAL_STATE }));
      })
      .catch(error => {
        this.setState({ error });
      });

    e.preventDefault();
  };

  render() {
    const { password, confirmPassword, error } = this.state;

    const isInvalid = password !== confirmPassword || password === '';

    return (
      <form onSubmit={this.onSubmit}>
        <input
          value={password}
          onChange={e => this.setState({ password: e.target.value })}
          type="password"
          placeholder="New Password"
        />
        <input
          value={confirmPassword}
          onChange={e => this.setState({ confirmPassword: e.target.value })}
          type="password"
          placeholder="Confirm New Password"
        />
        <button disabled={isInvalid} type="submit">
          Reset My Password
        </button>

        {error && <p>{error.message}</p>}
      </form>
    );
  }
}
