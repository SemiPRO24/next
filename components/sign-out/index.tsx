import React from 'react';

import { auth } from '../../firebase/index';

export const SignOutButton: React.FC<{}> = () => (
  <button type="button" onClick={auth.doSignOut}>
    Sign Out
  </button>
);
