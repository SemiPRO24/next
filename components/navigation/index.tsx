import React from 'react';
import Link from 'next/link';
import { connect } from 'react-redux';

import * as routes from '../../constants/routes';
import { SignOutButton } from '../sign-out';

type NavigationProps = {
  authUser: any;
};

const NavigationAuth = () => (
  <ul>
    <li>
      <Link href={routes.LANDING}>
        <a>Landing</a>
      </Link>
    </li>
    <li>
      <Link href={routes.HOME}>
        <a>Home</a>
      </Link>
    </li>
    <li>
      <Link href={routes.ACCOUNT}>
        <a>Account</a>
      </Link>
    </li>
    <li>
      <SignOutButton />
    </li>
  </ul>
);

const NavigationNonAuth = () => (
  <ul>
    <li>
      <Link href={routes.LANDING}>
        <a>Landing</a>
      </Link>
    </li>
    <li>
      <Link href={routes.SIGN_IN}>
        <a>Sign In</a>
      </Link>
    </li>
  </ul>
);

const NavigationComponent: React.FC<NavigationProps> = ({ authUser }) => (
  <div>{authUser ? <NavigationAuth /> : <NavigationNonAuth />}</div>
);

const mapStateToProps = (state: any) => ({
  authUser: state.sessionState.authUser,
});

export const Navigation = connect(mapStateToProps)(NavigationComponent);
