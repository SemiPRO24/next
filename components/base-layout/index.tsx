import React from 'react';
import Head from 'next/head';
import { compose } from 'recompose';

import withAuthentication from '../../hocs/withAuthentication';
import withAuthorization from '../../hocs/withAuthorization';
import { Navigation } from '../navigation';

type TBaseLayout = {
  className?: string;
  cannonical?: string;
  title?: string;
  children?: any;
};

const BaseLayout = (props: TBaseLayout) => {
  const { className, children, cannonical, title } = props;

  return (
    <React.Fragment>
      <Head>
        <title>{title || 'SV Next FA'}</title>
        <meta
          name="description"
          content="My name is Sviatoslav Nazar and I am an software engineer. I have a Master's degree in Applied mathematics and Informatics and several years of experience working as front-end developer."
        />
        <meta
          name="keywords"
          content="sviatoslav nazar, sviatoslav inoxoft, semipro, inoxoft"
        />
        <meta
          property="og:title"
          content="Sviatoslav Nazar - programmer, developer, gamer"
        />
        <meta property="og:locale" content="en_EU" />
        <meta property="og:url" content={`${process.env.BASE_URL}`} />
        <meta property="og:type" content="website" />
        <meta
          property="og:description"
          content="My name is Sviatoslav Nazar and I am an software engineer."
        />
        <link rel="manifest" href="/static/manifest.json" />
        <link
          href="https://fonts.googleapis.com/css?family=Montserrat:400,700"
          rel="stylesheet"
        />
        {cannonical && (
          <link
            rel="cannonical"
            href={`${process.env.BASE_URL}${cannonical}`}
          />
        )}
        <link rel="icon" type="image/ico" href="/static/favicon.ico" />
      </Head>
      <div className="layout-container">
        <Navigation />
        <main className={`cover ${className}`}>
          <div className="wrapper">{children}</div>
        </main>
      </div>
    </React.Fragment>
  );
};

const BaseLayoutWithAuthentication = compose(
  withAuthentication,
  withAuthorization(false)
)(BaseLayout);
const BaseLayoutWithAuthorization = compose(
  withAuthentication,
  withAuthorization(true)
)(BaseLayout);

export {
  BaseLayout,
  BaseLayoutWithAuthentication,
  BaseLayoutWithAuthorization,
};
