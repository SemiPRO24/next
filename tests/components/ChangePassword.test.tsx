import React from 'react';
import renderer from 'react-test-renderer';
import { ChangePassword } from '../../components/change-password';

describe('ChangePassword', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<ChangePassword />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
