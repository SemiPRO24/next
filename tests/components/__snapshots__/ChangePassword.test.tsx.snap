// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword should render correctly 1`] = `
<form
  className=""
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formChangePassword"
    >
      New Password
    </label>
    <input
      className="form-control"
      id="formChangePassword"
      onChange={[Function]}
      placeholder="New Password"
      type="password"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="formChangeConfirmPassword"
    >
      Confirm New Password
    </label>
    <input
      className="form-control"
      id="formChangeConfirmPassword"
      onChange={[Function]}
      placeholder="Confirm New Password"
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={true}
    type="submit"
  >
    Reset My Password
  </button>
</form>
`;
